File: agent/commands-generator-api/src/main/java/org/matilda/commands/types/DoubleConverter.java
package org.matilda.commands.types;

import com.google.protobuf.DoubleValue;

public class DoubleConverter implements ProtobufConverter<Double> {
    @Override
    public DoubleValue convert(Double object) {
        return DoubleValue.newBuilder().setValue(object).build();
    }
}

File: agent/commands-generator-api/src/main/java/org/matilda/commands/types/DoubleConverter.kt
package org.matilda.commands.types

import pbandk.unpack
import pbandk.wkt.Any
import pbandk.wkt.DoubleValue

class DoubleConverter : ProtobufConverter<Double> {
    override fun convertToProtobuf(obj: Double): DoubleValue {
        return DoubleValue(obj)
    }

    override fun convertFromProtobuf(obj: Any): Double {
        return obj.unpack(DoubleValue.Companion).value
    }
}
